packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -91,139 +121,548 @@ const propTypes = {
+    // check the length of inputValue to make sure that it is less then 2.
+    if (inputValue === date.month || inputValue.length > 2 || Number(inputValue) > 12 || inputValue === '00') {
+      return;
+    }
+
+    // If the change made was not a deletion of a digit, then prepend '0'
+    // when the input value is a single digit value between 2 and 9
+    if (inputValue.length >= date.month.length) {
+      const digitsToPrependZero = ['2', '3', '4', '5', '6', '7', '8', '9'];
+
+      if (digitsToPrependZero.indexOf(inputValue) > -1) {
+        inputValue = `0${inputValue}`;
+      }
+    }
+
+    handleDateChange(event, inputValue, DateUtil.inputType.MONTH);
+  };
+
+  const handleYearChange = (event) => {
+    const inputValue = event.target.value;
+    if (!DateUtil.validDateInput(inputValue)) {
+      return;
+    }
+
+    // Ignore the entry if the value did not change or it is invalid.
+    // When 'Predictive text' is enabled on Android the maxLength attribute on the input is ignored so we have to
+    // check the length of inputValue to make sure that it is less then 4.
+    if (inputValue === date.year || inputValue.length > 4) {
+      return;
+    }
+
+    // Ignore the entry if the year value is not between 1900 and 2100
+    if (inputValue.length === 4 && (Number(inputValue) < 1900 || Number(inputValue) > 2100)) {
+      return;
+    }
+
+    handleDateChange(event, inputValue, DateUtil.inputType.YEAR);
+  };
+
+  const setInputFocus = (event, inputRef, selectionStart, selectionEnd) => {
+    inputRef.focus();
+    inputRef.setSelectionRange(selectionStart, selectionEnd);
+    event.preventDefault();
+  };
+
+  const handleDayInputKeydown = (event) => {
+    if (inputAttributes?.readOnly) {
+      return;
+    }
+
+    if (dateFormatOrder === DateUtil.dateOrder.MDY) {
+      if ((event.keyCode === KEY_LEFT || event.keyCode === KEY_DELETE || event.keyCode === KEY_BACK_SPACE) && dayInputRef.current.selectionEnd === 0) {
+        setInputFocus(event, monthInputRef.current, date.month.length, date.month.length);
+      } else if (event.keyCode === KEY_RIGHT && dayInputRef.current.selectionEnd === date.day.length) {
+        setInputFocus(event, yearInputRef.current, 0, 0);
+      }
+    } else if (dateFormatOrder === DateUtil.dateOrder.DMY) {
+      if (event.keyCode === KEY_RIGHT && dayInputRef.current.selectionEnd === date.day.length) {
+        setInputFocus(event, monthInputRef.current, 0, 0);
+      }
+    } else if (dateFormatOrder === DateUtil.dateOrder.YMD) {
+      if ((event.keyCode === KEY_LEFT || event.keyCode === KEY_DELETE || event.keyCode === KEY_BACK_SPACE) && dayInputRef.current.selectionEnd === 0) {
+        setInputFocus(event, monthInputRef.current, date.month.length, date.month.length);
+      }
+    }
+  };
+
+  const handleMonthInputKeydown = (event) => {
+    if (inputAttributes?.readOnly) {
+      return;
+    }
+
```

Comment:
@avinashg1994  After reviewing - we'll need to add the additional KEY_UP and KEY_DOWN handlers as well for the Month, Day, and Year, to match the behavior in the Time-Input, which will be noticeable in the combined Date-Time-Picker if it works for time but not the date.

Will be similar to:
https://github.com/cerner/terra-framework/blob/9ffeb2c2f84b252e5bba42eba12738e9e1df3a70/packages/terra-time-input/src/TimeInput.jsx#L483-L489

\
\
Most likely should also add new Utils for the Incrementing & Decrementing, similar to Time-Input:
https://github.com/cerner/terra-framework/blob/9ffeb2c2f84b252e5bba42eba12738e9e1df3a70/packages/terra-time-input/src/TimeUtil.js#L100-L105

Month:
-Pressing Up/Down will cycle from 1-12.
\
 Day: 
 -Pressing Up/Down with no month & no year valued will cycle from 1-31.
-Pressing Up/Down with Month=02(February) & no year will cycle from 1-28.
-Pressing Up/Down with Month=02(February) & Year=LeapYear will cycle from 1-29.
-Pressing Up/Down with Month=04(April), 06(June), 09(September), 11(November) will cycle from 1-30.
 -If the day input already contains a number (e.g. 30), and then the month is valued which changes the valid range (e.g.1-28), pressing up would set it to 01 (the next available valid value), pressing down would change it to 28 (the next available valid value).   
\
Year:
-Pressing Up/Down will cycle between 1900 and 2100 (reading whatever the min-max range is currently set to).

Reply:
Added functionality to increment or decrement the input values using the Up or Down arrow keys: https://github.com/cerner/terra-framework/pull/1208/commits/1d83d4d33f91233ca936e2e9c19f4428246a32d8

Reply:
When using the arrow keys to update the month/day/year, The new date doesn't get updated in the calendar picker. It looks like onChange or onChangeRaw isn't being called. You can try using these tests:

- https://terra-framew-split-date-uw47dd.herokuapp.com/tests/terra-date-picker/date-picker/date-picker-on-change
- https://terra-framew-split-date-uw47dd.herokuapp.com/tests/terra-date-picker/date-picker/date-picker-on-change-raw

Reply:
The 'onChange' and 'onChangeRaw' issue has been addressed here:
https://github.com/cerner/terra-framework/pull/1208/commits/03ff5cd7030025dda29305ffdfe811c91fd54bf1